### DISCUSSION TITLE
MacOS image configuration PRIOR to the template in this repo?

### DISCUSSION THREAD

**Discussion by naikrovek** [actions/virtual-environments#7150] [Image configuration]
Hello, 

I'm adapting the MacOS Packer templates stored in this repo to work with Tart because I've already written a VM orchestration system using that tool (the templates here use Anka for local developer machine template development from what I can see) and I'm needing to make a lot of little tiny changes throughout the build in order to make things work.  Clearly I have some initial state wrong on the MacOS VM that I'm building on.

Is there any description of any customization on top of plain MacOS after the first-time setup is complete?  There's got to be something, but I can't work out what it is based on the errors that I'm seeing.

**Comment by mikhailkoliada** [comment]
@naikrovek Hello! The only thing I can think of is paswordless sudo and making sure your ssh port/service is available.

**Reply by naikrovek** [reply]
Thank you.  There has got to be something else, I think.  A lot of paths require tweaking in the tests as I build the image, and fix one error at a time before rebuilding and moving onto the next.  Is Homebrew stuff installed top /opt/homebrew/ or some other location?

There are definitely things that this automation assumes about the VM it starts in, or I would not be seeing the errors that I am seeing, I think.  I'm missing something somewhere.

**Reply by naikrovek** [reply]
What I'm starting with is the template in the link below.  I originally was using this template and this `tart` tool to do this, but now I do it manually because automating this phase of the VM image seems to break a lot.  I haven't looked into why.  

https://github.com/cirruslabs/macos-image-templates/blob/master/templates/vanilla-ventura.pkr.hcl

Here you can see the keystrokes it sends during the OS setup procedure, and then once those things are done, I SSH in and execute the commands shown in the second half of the packer template.

I'm missing something, though, because the MS/GitHub MacOS template doesn't apply cleanly to this image, even if I am very careful about changing the Packer builder from Anka (which I don't use) to `tart` (which I do use, for now.)

**Reply by PrashantRaj18198** [reply]
@naikrovek I'm working on a similar setup, were you able to resolve this?

**Reply by naikrovek** [reply]
no, I wound up getting pulled onto another project.
